src/data/polynomial/ring_division.lean

```diff
@@ -575,6 +575,14 @@ then if h : (X : R[X]) ^ n - C a = 0
 else by rw [← with_bot.coe_le_coe, ← degree_X_pow_sub_C (nat.pos_of_ne_zero hn) a];
   exact card_roots (X_pow_sub_C_ne_zero (nat.pos_of_ne_zero hn) a)
 
+@[simp]
+lemma nth_roots_two_eq_zero_iff {r : R} : nth_roots 2 r = 0 ↔ ¬ is_square r :=
+begin
+  simp only [nth_roots, eq_zero_iff_forall_not_mem, is_square, eval_sub, eval_pow, eval_X,
+    mem_roots (X_pow_sub_C_ne_zero two_pos r), sub_eq_zero, is_root.def, eval_C, not_exists],
+  simp [eq_comm, pow_two], -- simps cannot be combined as the order matters
+end
```

Comment:
```suggestion
by simp_rw [is_square_iff_exists_sq, eq_zero_iff_forall_not_mem,
            mem_nth_roots (by norm_num : 0 < 2), ← not_exists, eq_comm]
```
Making [multiset.exists_mem_of_ne_zero](https://leanprover-community.github.io/mathlib_docs/data/multiset/basic.html#multiset.exists_mem_of_ne_zero) an iff could also make the proof slightly easier.